tests/unittests/tests-fib_sr/tests-fib_sr.c

```diff
@@ -0,0 +1,701 @@
+                          (uint8_t *)&addr_nxt, add_buf_size,
+                          &sr_path_entry[0])
+                         );
+
+    /* now we go to a earler entry and add a new entry not dropping the remaining ones */
+    snprintf(addr_nxt, add_buf_size, "Some address X2");
+
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_search(local_sourceroutes[0],
+                                           (uint8_t *)&addr_nxt, add_buf_size,
+                                           &sr_path_entry[0])
+                         );
+
+    snprintf(addr_nxt, add_buf_size, "Some address XZ");
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_entry_add(local_sourceroutes[0], sr_path_entry[0],
+                          (uint8_t *)&addr_nxt,
+                          add_buf_size, true)
+                         );
+
+    /*
+    * @note uncomment the following line to print the source route.
+    * expected X0, .., X2, XZ, X3, .., X7, XY
+    */
+    //fib_print_sr( local_sourceroutes[0]);
+
+    /* remove the source route */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_delete(local_sourceroutes[0]));
+
+    fib_deinit(NULL);
+}
+
+
+/*
+* @brief create a new source route with a number of hops on its path,
+*        and delete several entries
+*/
+static void test_fib_sr_07_create_sr_with_hops_and_delete_one(void)
+{
+    fib_sr_t *local_sourceroutes[1];
+    size_t add_buf_size = 16;
+    char addr_nxt[add_buf_size];
+
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_create(&local_sourceroutes[0], 42, 0x0, 10000));
+
+    for (size_t i = 0; i < 10; ++i) {
+        snprintf(addr_nxt, add_buf_size, "Some address X%d", i);
+        /* append a  hop */
+        TEST_ASSERT_EQUAL_INT(0, fib_sr_entry_append(local_sourceroutes[0],
+                              (uint8_t *)&addr_nxt,
+                              add_buf_size)
+                             );
+    }
+
+    snprintf(addr_nxt, add_buf_size, "Some address X6");
+    /* we delete X6 keeping the remaining entries */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_entry_delete(local_sourceroutes[0],
+                          (uint8_t *)&addr_nxt, add_buf_size,
+                          true)
+                         );
+
+    snprintf(addr_nxt, add_buf_size, "Some address X9");
+    /* now the destination */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_entry_delete(local_sourceroutes[0],
+                          (uint8_t *)&addr_nxt, add_buf_size,
+                          true)
+                         );
+
+    snprintf(addr_nxt, add_buf_size, "Some address X0");
+    /* and now the first hop on the path */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_entry_delete(local_sourceroutes[0],
+                          (uint8_t *)&addr_nxt, add_buf_size,
+                          true)
+                         );
+
+    /*
+    * @note uncomment the following line to print the source route.
+    * expected X1, .., X5, X7, .., X8
+    */
+    //fib_print_sr( local_sourceroutes[0]);
```

Comment:
instead of this commented `fib_print_sr` line, you could actually `TEST_ASSERT` those entries with a loop (: